lib/cookies.js

```diff
@@ -70,130 +56,131 @@ CookieJar.prototype.add = function(name, value, expiry, path, domain, httponly,
 
     // Are we updating an existing cookie or adding a new one?
     this.cookies.forEach(function(cookie, index) {
-        if (cookie.name === newCookie.name &&
-            cookie.matchDomain(newCookie.domain)) {
-
+        if (cookie.name === newCookie.name && cookie.matchDomain(newCookie.domain)) {
             existingIndex = index;
         }
     });
 
-    if (existingIndex < 0) {
+    if (existingIndex === -1) {
         this.cookies.push(newCookie);
     } else {
         this.cookies[existingIndex] = newCookie;
     }
 
+    /**
+     * Fired when a cookie has been added to the jar
+     * @event CookieJar#addcookie
+     * @param {Cookie} cookie The cookie that has been added
+     */
     this.emit("addcookie", newCookie);
 
-    if (cb && cb instanceof Function) {
-        cb(null, newCookie);
+    if (callback instanceof Function) {
+        callback(null, newCookie);
     }
 
     return this;
 };
 
-/*
-    Public: Removes cookies from the cookie jar. If no domain and name are
-    specified, all cookies in the jar are removed.
-
-    name                -   The name of the cookie(s) to remove
-    domain              -   The domain from which to remove cookies.
-    cb                  -   Optional callback.
-
-    Emits
-
-        removecookie    -   Emitted with array of removed cookies.
-
-    Examples
-
-        cookieJar.remove(null,"nytimes.com");
-
-    Returns an array of removed cookies.
+/**
+ * Called when {@link CookieJar#remove} returns a result
+ * @callback CookieJar~removeCallback
+ * @param {Error|null} error             If the operation was successful, this will be `null`. Otherwise it will be the error that was encountered.
+ * @param {Cookie[]|null} cookiesRemoved An array of the cookies that were removed from the cookie jar
+ */
 
-*/
-CookieJar.prototype.remove = function(name, domain, cb) {
+/**
+ * Removes cookies from the cookie jar. If no domain and name are specified, all
+ * cookies in the jar are removed.
+ * @fires CookieJar#removecookie
+ * @param {String} [name]                       Name of the cookie to be removed
+ * @param {String} [domain]                     The domain that the cookie applies to
+ * @param {CookieJar~removeCallback} [callback]
+ * @return {Cookie[]}                           Returns an array of the cookies that were removed from the cookie jar
+ */
+CookieJar.prototype.remove = function(name, domain, callback) {
     var cookiesRemoved = [],
         jar = this;
 
     jar.cookies.forEach(function(cookie, index) {
-
         // If the names don't match, we're not removing this cookie
-        if (Boolean(name) && cookie.name !== name) {
+        if (typeof name === "string" && cookie.name !== name) {
```

Comment:
FWIW I kinda read this line as "Ah, we're testing it to see if it's a string, so it's probably a polymorphic variable." That said though, it's pretty clearly documented, so I don't see any problem with continuing as is. :)